### PR TITLE
runtime: fix stability feature flags for docs

### PR DESCRIPTION
**Pull request by joshka**
## Motivation

Fixes: #6907

## Solution

Adds doc attribute noting the tokio_unstable requirements for various items

### CODE REVIEW

File: tokio/src/runtime/task/mod.rs [hunk 1]
```diff
@@ -391,7 +391,6 @@ impl<S: 'static> Task<S> {
         ///
         /// [task ID]: crate::task::Id
         #[cfg(tokio_unstable)]
-        #[cfg_attr(docsrs, doc(cfg(tokio_unstable)))]
```
Comment (joshka) [inline tokio/src/runtime/task/mod.rs:394]: This has no effect as the method is pub(crate)

File: tokio/src/lib.rs [hunk 2]
```diff
@@ -351,8 +351,10 @@
 //! - [`task::Builder`]
 //! - Some methods on [`task::JoinSet`]
 //! - [`runtime::RuntimeMetrics`]
+//! - [`runtime::Builder::on_task_spawn`]
+//! - [`runtime::Builder::on_task_terminate`]
 //! - [`runtime::Builder::unhandled_panic`]
-//! - [`task::Id`]
```
Comment (joshka) [inline tokio/src/lib.rs:355]: This was stabilized in https://github.com/tokio-rs/tokio/pull/6891

File: tokio/src/runtime/runtime.rs [hunk 3]
```diff
@@ -119,6 +119,7 @@ pub enum RuntimeFlavor {
     MultiThread,
     /// The flavor that executes tasks across multiple threads.
     #[cfg(tokio_unstable)]
+    #[cfg_attr(docsrs, doc(cfg(tokio_unstable)))]
```
Comment (joshka) [inline tokio/src/runtime/runtime.rs:122]: This was not mentioned in the related issue, but I did a check for all the tokio_unstable marked items across the repo and this one was the only one that was obviously missing.

**Review by joshka — COMMENTED** [review]
As a sidenote, I wonder if it would be possible to use https://doc.rust-lang.org/rustdoc/unstable-features.html#doc_auto_cfg-automatically-generate-doccfg instead of manually adding all these attributes? Are there places where you would not want the specified items to be documented with their required feature flags? I did see a couple of places where there's a different implementation depending on which flag is in use, perhaps that's the problem here?

File: tokio/src/runtime/builder.rs [hunk 4]
```diff
@@ -728,6 +728,7 @@ impl Builder {
     /// # }
     /// ```
     #[cfg(all(not(loom), tokio_unstable))]
+    #[cfg_attr(docsrs, doc(cfg(all(not(loom), tokio_unstable))))]
```
Comment (Darksonn) [inline tokio/src/runtime/builder.rs:731]: Loom should not be mentioned in docs.
```suggestion
    #[cfg_attr(docsrs, doc(cfg(tokio_unstable)))]
```

**Review by Darksonn — COMMENTED** [review]
Unfortunately auto cfg docs only works for features and not for `--cfg` flags like `tokio_unstable`.

File: tokio/src/runtime/builder.rs (hunk 4 above)
Comment (Darksonn) [reply tokio/src/runtime/builder.rs]: Please fix all instances of this issue. Marking unresolved.

File: tokio/src/runtime/builder.rs [hunk 5]
```diff
@@ -728,6 +728,7 @@ impl Builder {
     /// # }
     /// ```
     #[cfg(all(not(loom), tokio_unstable))]
+    #[cfg_attr(docsrs, doc(cfg(tokio_unstable)))]
     pub fn on_task_spawn<F>(&mut self, f: F) -> &mut Self
```
Comment (Darksonn) [inline tokio/src/runtime/builder.rs:732]: Please also add the docs blurb about this being unstable.
```rs
/// **Note**: This is an [unstable API][unstable]. The public API of this type
/// may break in 1.x releases. See [the documentation on unstable
/// features][unstable] for details.
///
/// [unstable]: crate#unstable-features
```
I would add it to `on_task_spawn`, `on_task_terminate`, and `TaskMeta`. Probably not to `MultiThreadAlt`.

Comment (joshka) [reply tokio/src/runtime/builder.rs:738]: Done in 6b438c15

File: tokio/src/runtime/builder.rs (hunk 4 above)
Comment (joshka) [reply tokio/src/runtime/builder.rs]: Done in 7aaac8a4

**Review by Darksonn — APPROVED** [review]
Thank you.

**Pull request merged**